libraries/stdlib/src/kotlin/CharCode.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package kotlin
+
+
+/**
+ * Creates a Char with the specified [code], or throws an exception if the [code] is out of `Char.MIN_VALUE.code..Char.MAX_VALUE.code`.
+ *
+ * If the program that calls this function is written in a way that only valid [code] is passed as the argument,
+ * using the overload that takes a [UShort] argument is preferable (`Char(intValue.toUShort())`).
+ * That overload doesn't check validity of the argument, and may improve program performance when the function is called routinely inside a loop.
+ *
+ * @sample samples.text.Chars.charFromCode
+ */
+@OptIn(ExperimentalUnsignedTypes::class)
+@ExperimentalStdlibApi
+@SinceKotlin("1.4")
+@kotlin.internal.InlineOnly
+public inline fun Char(code: Int): Char {
+    if (code < Char.MIN_VALUE.code || code > Char.MAX_VALUE.code) {
```

Comment:
If future, I'd extract this check to a helper internal function like following:
`internal fun Int.checkCharCodeRange(): Int`, so that this body would look like
`return Char(code.checkCharCodeRange().toUShort())`

Reply:
The check can also be simplified down to one comparison by checking that the upper 16 bits are all zero.